### PR TITLE
Adding support for jenkins config

### LINKED ISSUE #42: Load jenkins system configuration
### Description
Configuring jenkins is a complex process. Currently, we configure jenkins using the UI which cannot be replicated easily. We need a method to be able to load and change the jenkins configuration through a configuration file.

### Acceptance Criteria
CDK deployment deploys a configuration file which is used to configure jenkins automatically during deployment.

### PR DESCRIPTION
**Pull request by abhinavGupta16**
Signed-off-by: Abhinav Gupta <abhng@amazon.com>

### Description
- Added support for loading jenkins through config file `jenkins.yaml`
- Sorted the list of plugins for easy searching
 
### Issues Resolved
#42 

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins.yaml [hunk 1]
```diff
@@ -0,0 +1,229 @@
+jenkins:
```
[the remark this answers is not in the document]
Comment (abhinavGupta16) [reply jenkins.yaml:1]: I picked it out from the existing jenkins that the cdk deploys. We can edit it to our needs and changes will deploy automatically.

File: jenkins.yaml [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    okToTestPhrase: ".*ok\\W+to\\W+test.*"
+    retestPhrase: ".*test\\W+this\\W+please.*"
+    skipBuildPhrase: ".*\\[skip\\W+ci\\].*"
+    useComments: false
+    useDetailedComments: false
+    whitelistPhrase: ".*add\\W+to\\W+whitelist.*"
+  gitHubConfiguration:
+    apiRateLimitChecker: ThrottleForNormalize
+  gitHubPluginConfig:
+    hookUrl: "http://ci-de-jenki-18z6t74yxghfp-1257712944.us-east-1.elb.amazonaws.com/github-webhook/"
```
Comment (gaiksaya) [inline jenkins.yaml:189]: Not sure how this is generated but is this variable and needs to be set?

Comment (abhinavGupta16) [reply jenkins.yaml:189]: Webhooks are a a part of settings in github. We don't have that set. I am not sure either.

Comment (gaiksaya) [reply jenkins.yaml:189]: Looks like an ELB endpoint. @peterzhuamazon any idea about this?

File: lib/compute/jenkins-plugins.ts [hunk 3]
```diff
@@ -7,150 +7,149 @@
  */
 
 export class JenkinsPlugins {
-    // Below are the list of plugins and their dependencies currently in use in prod
+    // Below are the list of plugins (sorted alphabetically) and their dependencies currently in use in prod
     static readonly plugins: string[] = [
```
Comment (peternied) [inline lib/compute/jenkins-plugins.ts:11]: Add a unit tests to make sure these are always in sorted order

**Review by peternied — CHANGES_REQUESTED** [review]

**Review by peternied — APPROVED** [review]

File: jenkins.yaml [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    maxAttachmentSize: -1
+    maxAttachmentSizeMb: -1
+    precedenceBulk: false
+    watchingEnabled: false
+  fingerprints:
+    fingerprintCleanupDisabled: false
+    storage: "file"
+  ghprbTrigger:
+    autoCloseFailedPullRequests: false
+    cron: "H/5 * * * *"
```
Comment (peterzhuamazon) [inline jenkins.yaml:170]: Do not add cron for ghprb in global settings.
Remove it or make it blank so workflows can set their own.

File: jenkins.yaml [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    extensions:
+    - ghprbSimpleStatus:
+        addTestResults: false
+        showMatrixStatus: false
+    githubAuth:
+    - description: "Anonymous connection"
+      id: "fe7046c5-29a0-4370-b32d-47aead7c5945"
```
Comment (peterzhuamazon) [inline jenkins.yaml:177]: Think ID is auto generated not hard coded?

File: jenkins.yaml (hunk 2 above)
Comment (peterzhuamazon) [reply jenkins.yaml:189]: Auto set by plugin no need to add it I think.

**Review by gaiksaya — APPROVED** [review]
First pass. LGTM

**Pull request merged**